### PR TITLE
Move amount

### PR DESCRIPTION
**Pull request by lukefx**
I'm working on the possibility to move a subset of items, the UI is there but I'm having hard times on understanding your code. You should really add some comments...for guardian's sake :smile: 

image

This already works, it moves the right number of items, but then I have to reload the page because the source disappears...

### CODE REVIEW

**Comment by SunburnedGoose** [comment]
I was imagining more features on the move dialog.  

Take all button.  
Take one button.  
Increment, decrement operations.  

I also have to change the move operation when there are items left behind, I don't remove the source.

**Comment by lukefx** [comment]
If you don't type a number it will take all, we could found a way for the
take one. There is a numeric stepper so you don't have to type any number
if you don't want.
This could be a start...then we could iterate and add all sort of
functionality.
Another idea is to create a radial menu instead of the usual pop-up.

**Comment by SunburnedGoose** [comment]
FYI, It also needs to support drag and drop too.

**Comment by lukefx** [comment]
Couldn't we suppose that if you drag the icon...you want it all like does now?

**Comment by SunburnedGoose** [comment]
I would prefer to be prompted when moving stackable items that are more than one.  Even if that isn't the default, I would want to make it an option.

File: app/scripts/move-popup/dimMovePopup.directive.js [hunk 1]
```diff
@@ -95,6 +96,14 @@
 
     vm.stores = dimStoreService.getStores();
 
+    this.canShowAmount = function(item, itemStore, buttonStore) {
+      var result = false;
+      if (item.amount > 1) {
+        result = true;
+      }
+      return result;
+    };
```
Comment (kyleshay) [inline app/scripts/move-popup/dimMovePopup.directive.js]: why not just 
    return item.amount > 1;

Comment (lukefx) [inline app/scripts/move-popup/dimMovePopup.directive.js]: yepp absolutely. the function has wrong arity too...the moveAmount functionality is not completed and this is poor copy and paste code...shame on me :finnadie:

**Comment by rduenasf** [comment]
you could do it as some games do it, if you hold shift while drag-and-drop-ing it'll ask you for the amount that you want, otherwise it just assumes you're moving everything.

Ref: search for shiftKey in https://developer.mozilla.org/en-US/docs/Web/Events/dragend

**Comment by SunburnedGoose** [comment]
Might want to make a toaster that lets users know that it is an option.

**Comment by Subject22** [comment]
I'd also suggest graphically modifying stacks whenever the shift key is held to indicate the difference.

**Comment by jflayhart** [comment]
I like @Subject22 idea... Can hold down shift and then on drop a dialog comes up asking how many items you want to move from that operation.

**Comment by SunburnedGoose** [comment]
How does that translate to a mobile device?

**Comment by lukefx** [comment]
On mobile i think that a good solution could be, if you hold the tap on materials...it will pop a slider and let you choose the amount. Something like this (http://skidding.github.io/dragdealer) horizontally.
What do you think?

**Comment by SunburnedGoose** [comment]
We have to ways to move things in v4: single item and mult-select.  The issue is if we move multiple items, how do we let the user choose all or partial?  I want to reserve our gestures for other things like viewing details (long press).

I was thinking that when we move a stackable item, we always prompt for amount.

**Comment by vincchan** [comment]
I have not worked out how this would play out on mobile but how about instead of cluttering up the interface with more buttons, slider and dialog, we do this:

* Detect when a player right clicks on inventory item with more than one amount
* Split the amount (for example if I had 44 x 3oC I would move 22)
* Have the Icon with 22 x 3oC follow the curser to drop it somewhere

Something along of the lines like they do it in minecraft

screen recording 2015-10-10 at 09 10 am

**Comment by kyleshay** [comment]
closing this PR, still needs some thought and development.

**Pull request closed without merging**